tools/stylelint/no-concrete-rules/index.js

```diff
@@ -0,0 +1,39 @@
+const stylelint = require('stylelint');
+const path = require('path');
+const ruleName = 'material/no-concrete-rules';
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  expected: pattern => `CSS rules must be placed inside a mixin for files matching '${pattern}'.`
+});
+
+/**
+ * Stylelint plugin that will log a warning for all top-level CSS rules.
+ * Can be used in theme files to ensure that everything is inside a mixin.
+ */
+const plugin = stylelint.createPlugin(ruleName, (isEnabled, options) => {
+  return (root, result) => {
+    if (!isEnabled) return;
+
+    const filePattern = new RegExp(options.filePattern);
+    const fileName = path.basename(root.source.input.file);
+
+    if (!filePattern.test(fileName)) return;
+
+    // Go through all the nodes and report a warning for every CSS rule.
+    // We use a regular `forEach`, instead of the PostCSS walker utils,
+    // because we only care about the top-level nodes.
+    root.nodes.forEach(node => {
+      if (node.type === 'rule') {
```

Comment:
Could this also catch doing `@include some-mixin()` in a partial?

Reply:
It could, but I'm not sure whether we want that. [We have some cases](https://github.com/angular/material2/blob/master/src/lib/datepicker/_datepicker-theme.scss#L12) where we use mixins inside the theme files to reduce duplication. We could make an exception for mixins that start with an underscore though.

Reply:
I mean specifically at the root level; `@include` inside of another mixin would be fine.

Reply:
Done.